src/opaque.rs

```diff
@@ -983,11 +491,45 @@ impl<CS: CipherSuite> ClientLogin<CS> {
     }
 }
 
-type ClientLoginFinishResult<CS> = (
-    LoginThirdMessage<CS>,
-    Vec<u8>,
-    GenericArray<u8, ExportKeySize>,
-);
+/// Optional parameters for client login start
+pub enum ClientLoginStartParameters {
```

Comment:
nit: this could actually implement (and use) [`core::default::Default`](https://doc.rust-lang.org/core/default/trait.Default.html)